xline/src/server/maintenance.rs

```diff
@@ -67,9 +81,32 @@ where
         &self,
         _request: tonic::Request<StatusRequest>,
     ) -> Result<tonic::Response<StatusResponse>, tonic::Status> {
-        Err(tonic::Status::unimplemented(
-            "status is unimplemented".to_owned(),
-        ))
+        let cluster = self
+            .client
+            .get_cluster_from_curp(false)
+            .await
+            .map_err(propose_err_to_status)?;
+        let header = self.header_gen.gen_header();
+        let self_id = self.cluster_info.self_id();
+        let is_learner = cluster
+            .members
+            .iter()
+            .find(|member| member.id == self_id)
+            .map_or(false, |member| member.is_learner);
+        // TODO should we add `raft_index` and `raft_applied_index` in FetchClusterResponse?
```

Comment:
Same as above, Could we add a set of interfaces to the curp server to get state externally

Reply:
I think this interface is similar to `fetch cluster`. We can add more fields to the fetch cluster response.

Reply:
After bypassing the RPC sent to oneself within the curp client, more detailed information can be obtained using the curp node inside the client. I will handle this in the upcoming PR.